java/src/jmri/jmrit/withrottle/MultiThrottleController.java

```diff
@@ -307,6 +308,27 @@ else if ( question == DecisionType.STEAL_OR_SHARE ){ // using the same process a
         
     }
 
+    // Encode a SpeedStepMode to a string.
+    private static String encodeSpeedStepMode(SpeedStepMode mode) {
+        switch(mode) {
+            // NOTE: old speed step modes use the original numeric values
+            // from when speed step modes were in DccThrottle. New speed step
+            // modes use the mode name.
```

Comment:
I'm not very happy with this, but I suspect it's the only way to maintain protocol compatibility. I didn't see any documentation for this protocol, is it possible to change it?

Reply:
The protocol is documented at http://www.jmri.org/help/en/package/jmri/jmrit/withrottle/Protocol.shtml it’s not possible to change it. A number of products with their own release cycles implement it both as servers and clients.

Reply:
The class you are looking at here basically is the definition of the protocol.


Because there are devices in the field using this, we really can’t change the values sent here.

Reply:
I use the iPhone WiThrottle on my layout, so I'm not surprised that the protocol is difficult to change. Hacks it is.